### PR TITLE
<enhance>add deviceshifu-socket ut

### PR DESCRIPTION
**Pull request by MrLeea-13155bc**
**What this PR does / why we need it**:
add command handle and telemetry ut
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [X] unit test
- [ ] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/deviceshifu/deviceshifusocket/deviceshifusocket_test.go [hunk 1]
```diff
@@ -126,3 +134,169 @@ func TestEncodeMessage(t *testing.T) {
...
+
+	listener, err := net.Listen("tcp", "localhost:3000")
+	if err != nil {
+		t.Errorf("Cannot Listen at port 3000")
+	}
+
+	go func() {
+		_, _ = listener.Accept()
+	}()
+	// testcase pass
```
Comment (kris21he) [inline pkg/deviceshifu/deviceshifusocket/deviceshifusocket_test.go:164]: I suggest we adopt table driven unit test if we can.
please refer to https://dev.to/boncheff/table-driven-unit-tests-in-go-407b

The benefit of doing this is:
1. structured unit test
2. easy to maintain unit test

in some case if the test case is pretty simple to manage, we can ignore this practice

File: pkg/deviceshifu/deviceshifusocket/deviceshifusocket_test.go [hunk 2]
```diff
@@ -126,3 +152,256 @@ func TestEncodeMessage(t *testing.T) {
...
+			name:      "case1 not set header 'content-Type'",
+			request:   dc.Post(),
+			expErrStr: "the server rejected our request for an unknown reason",
+		},
+		{
+			name:      "case2 request Body is empty",
+			request:   dc.Post().SetHeader("Content-Type", "application/json"),
+			expErrStr: "the server rejected our request for an unknown reason",
+		}, {
+			name:      "case3 requestBody Encode is error",
```
Comment (tomqin93) [inline pkg/deviceshifu/deviceshifusocket/deviceshifusocket_test.go:329]: let's add a comment on why this should fail

**Review by tomqin93 — COMMENTED** [review]
@MrLeea-13155bc added 1 comment, the rest looks good

**Review by tomqin93 — APPROVED** [review]
LGTM

**Review by kris21he — APPROVED** [review]

**Pull request merged**